src/librustc_typeck/astconv.rs

```diff
@@ -269,12 +275,19 @@ impl<'o, 'gcx: 'tcx, 'tcx> AstConv<'gcx, 'tcx>+'o {
         let substs = Substs::for_item(tcx, def_id, |param, substs| {
             match param.kind {
                 GenericParamDefKind::Lifetime => {
-                    let i = param.index as usize - own_self;
-                    if let Some(lifetime) = parameters.lifetimes.get(i) {
```

Comment:
Have you tested what happens if you replace the `.get(i)` with indexing? Do all the tests pass?

Reply:
Oh I guess there's some checks above involving `lt_provided`.